include/oneapi/dpl/pstl/glue_memory_impl.h

```diff
@@ -280,15 +272,14 @@ oneapi::dpl::__internal::__enable_if_execution_policy<_ExecutionPolicy>
 uninitialized_default_construct(_ExecutionPolicy&& __exec, _ForwardIterator __first, _ForwardIterator __last)
 {
     typedef typename std::iterator_traits<_ForwardIterator>::value_type _ValueType;
-    typedef std::decay_t<_ExecutionPolicy> _DecayedExecutionPolicy;
 
     if constexpr (!std::is_trivially_default_constructible_v<_ValueType>)
     {
         const auto __dispatch_tag = oneapi::dpl::__internal::__select_backend(__exec, __first);
 
         oneapi::dpl::__internal::__pattern_walk1(
             __dispatch_tag, std::forward<_ExecutionPolicy>(__exec), __first, __last,
-            oneapi::dpl::__internal::__op_uninitialized_default_construct<_DecayedExecutionPolicy>{});
+            oneapi::dpl::__internal::__op_uninitialized_default_construct<std::decay_t<_ExecutionPolicy>>{});
```

Comment:
As far as we are changing template parameters of `__op_uninitialized_default_construct`....
It looks `__op_uninitialized_default_construct`has only one implementation for the host and device purposes. So, let remove `_ExecutionPolicy` at all.

Reply:
I don't know from implementation point of view how to do this.
@MikeDvorskiy could you please prepare some separate PR with this fix?